local_pathfinding/node_navigate.py

```diff
@@ -176,7 +178,22 @@ def get_desired_heading(self) -> float:
         )
 
         # TODO: create function to compute the heading from current position to next local waypoint
-        return 0.0
+        # Find index of self.gps in current waypoints
+        current_waypoint = self.gps.lat_lon
+        next_waypoint = None
+        waypoints = self.local_path.waypoints
+        for i in range(len(waypoints)):
+            if waypoints[i] == current_waypoint and i != len(waypoints) - 1:
+                next_waypoint = waypoints[i+1]
+                break
+        if next_waypoint is None:
+            return 0.0
+        heading = math.atan2(
+            (next_waypoint.longitude-current_waypoint.longitude) *
+            math.cos(current_waypoint.latitude),
+            (next_waypoint.latitude-current_waypoint.latitude)
+        )
```

Comment:
Mathematically this looks like it will work! You can improve the readability and accuracy by converting the latlons to XY using methods in `coord_systems.py`, and then do trigonometry on the XY components. You could also use `GEODESIC.inv()` method directly.

Reply:
Geodesic.inv [documentation](https://pyproj4.github.io/pyproj/stable/api/geod.html#pyproj.Geod.inv)

I'd suggest using named arguments, ie `geod.inv(lats1=10, ...)`, since the order of the arguments is kinda unintuitive. Normally I think latitude should come first (latitude, longitude), but their function flips this order.

You can also check `coord_systems.py` for example usage